### PR TITLE
V14 fixes (feat. Ben's quality memes)

### PR DESCRIPTION
**Pull request by sbrugel**
a "LGTM 👍" should do (once everything below is addressed)

**NOTES**:
- prune is commented out entirely due to an unfathomable amount of errors; not like we really need it at the moment anyways
- help: needs reworking (specifically for splitting) **(THIS HAS BEEN FIXED)**
- activity/status don't work
- serverlog randomly emits errors
- issue, refresh, commit untested

### CODE REVIEW

**Review by BenSegal855 — COMMENTED** [review]
Overall its not bad. A few recommendations here and there. Most notibly of all...
image

File: src/commands/admin/issue.ts [hunk 1]
```diff
@@ -2,36 +2,36 @@ import { ADMIN_PERMS } from '@lib/permissions';
...
+		type: ApplicationCommandOptionType.String,
 		required: false
 	}]
 
-	async run(interaction: CommandInteraction): Promise<void> {
-		const title = interaction.options.getString('title');
-		const label = interaction.options.getString('labels');
-		const body = interaction.options.getString('body');
+	async run(interaction: CommandInteraction): Promise<InteractionResponse<boolean> | void> {
+		const title = (interaction.options as CommandInteractionOptionResolver).getString('title');
```
[the remark this answers is not in the document]
Comment (craftablescience) [reply src/commands/admin/issue.ts:32]: `if (!interaction.isChatInputCommand()) return;`
Will unlock all the various methods on options

**Review by renross — APPROVED** [review]

**Comment by renross** [comment]
hotfix - disregard ts-ignore lines atm

File: src/commands/admin/issue.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sbrugel) [reply src/commands/admin/issue.ts]: thanks - this has been changed in every command file.

File: src/commands/admin/addcourse.ts [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
-	async run(interaction: CommandInteraction): Promise<void> {
+	async run(interaction: CommandInteraction): Promise<InteractionResponse<boolean> | void> {
 		interaction.reply('<a:loading:755121200929439745> working...');
 
-		const course = interaction.options.getString('course');
+		const course = (interaction.options as CommandInteractionOptionResolver).getString('course');
 		//	make sure course does not exist already
 		if (await interaction.client.mongo.collection(DB.COURSES).countDocuments({ name: course }) > 0) {
-			interaction.reply({ content: `${course} has already been registered as a course.`, ephemeral: true });
+			interaction.editReply({ content: `${course} has already been registered as a course.` });
```
[the remark this answers is not in the document]
Comment (sbrugel) [reply src/commands/admin/addcourse.ts:28]: Replying to this (and most of the others) for completeness: Probably something to do with there previously being a reply sent (the loading message). Most likely it was some weird thing with upgrading to 14.

File: src/commands/fun/8ball.ts
[the remark this answers is not in the document]
Comment (sbrugel) [reply src/commands/fun/8ball.ts:1]: setImage won't work on local paths. I could upload it to imgur or something and link it, though.

File: src/pieces/logs/serverLog.ts [hunk 3]
```diff
@@ -168,43 +170,43 @@ async function processChannelUpdate(oldChannel: GuildChannel | DMChannel, newCha
...
 async function processEmojiUpdate(oldEmote: GuildEmoji, newEmote: GuildEmoji, serverLog: TextChannel): Promise<void> {
 	if (newEmote.guild.id !== GUILDS.MAIN || newEmote.name === oldEmote.name) return;
-	const logs = (await newEmote.guild.fetchAuditLogs({ type: 'EMOJI_UPDATE', limit: 1 })).entries;
+	const logs = (await newEmote.guild.fetchAuditLogs({ type: AuditLogEvent.EmojiUpdate, limit: 1 })).entries;
 	const [logEntry] = [...logs.values()];
 
-	const embed = new MessageEmbed()
-		.setAuthor(logEntry.executor.tag, logEntry.executor.avatarURL({ dynamic: true }))
+	const embed = new EmbedBuilder()
+		.setAuthor({ name: logEntry.executor.tag, iconURL: logEntry.executor.avatarURL({ }) })
```
[the remark this answers is not in the document]
Comment (sbrugel) [reply src/pieces/logs/serverLog.ts:207]: I must have rushed through this, they have no params now.

**Comment by sbrugel** [comment]
Most fixes here are pushed to `help-fixes` branch, will be merged to `v14-fixes` tomorrow

**Review by bladetyr — APPROVED** [review]
never put that in the title again

**Review by bladetyr — APPROVED** [review]

**Review by bladetyr — DISMISSED** [review]

**Review by bladetyr — DISMISSED** [review]

**Review by mattnadar — APPROVED** [review]

**Review by leo-cx — APPROVED** [review]

**Review by mattnadar — APPROVED** [review]

**Review by bladetyr — APPROVED** [review]

**Pull request merged**